cmd/manager/main.go

```diff
@@ -0,0 +1,187 @@
+	log.Info(fmt.Sprintf("Version of operator-sdk: %v", sdkVersion.Version))
+}
+
+func main() {
+	// TODO: the operator-sdk switched to a new log implementation.
+	//       should we migrate the shared logger code?
+
+	// Add the zap logger flag set to the CLI. The flag set must
+	// be added before calling pflag.Parse().
+	pflag.CommandLine.AddFlagSet(zap.FlagSet())
+
+	// Add flags registered by imported packages (e.g. glog and
+	// controller-runtime)
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+
+	pflag.Parse()
+
+	// Use a zap logr.Logger implementation. If none of the zap
+	// flags are configured (or if the zap flag set is not being
+	// used), this defaults to a production zap logger.
+	//
+	// The logger instantiated here can be changed to any logger
+	// implementing the logr.Logger interface. This logger will
+	// be propagated through the whole operator, generating
+	// uniform and structured logs.
+	logf.SetLogger(zap.Logger())
+
+	printVersion()
+
+	namespace, err := k8sutil.GetWatchNamespace()
+	if err != nil {
+		log.Error(err, "Failed to get watch namespace")
+		os.Exit(1)
+	}
+
+	// Get a config to talk to the apiserver
+	cfg, err := config.GetConfig()
+	if err != nil {
+		log.Error(err, "")
+		os.Exit(1)
+	}
+
+	ctx := context.TODO()
+
+	// Become the leader before proceeding
+	err = leader.Become(ctx, "kubedirector-lock")
+	if err != nil {
+		log.Error(err, "")
+		os.Exit(1)
+	}
+
+	// Create a new Cmd to provide shared dependencies and start components
+	mgr, err := manager.New(cfg, manager.Options{
+		Namespace:          namespace,
+		MapperProvider:     restmapper.NewDynamicRESTMapper,
+		MetricsBindAddress: fmt.Sprintf("%s:%d", metricsHost, metricsPort),
+	})
+	if err != nil {
+		log.Error(err, "")
+		os.Exit(1)
+	}
+
+	log.Info("Registering Components.")
+
+	// Setup Scheme for all resources
+	if err := apis.AddToScheme(mgr.GetScheme()); err != nil {
+		log.Error(err, "")
+		os.Exit(1)
+	}
+
+	// Setup all Controllers
+	if err := controller.AddToManager(mgr); err != nil {
+		log.Error(err, "")
+		os.Exit(1)
+	}
+
+	// Create Service object to expose the metrics port.
+	_, err = metrics.ExposeMetricsPort(ctx, metricsPort)
+	if err != nil {
+		log.Info(err.Error())
+	}
+
+	// Fetch our deployment object
+	kdName, err := k8sutil.GetOperatorName()
+	if err != nil {
+		log.Error(err, "failed to get kubedirector deployment name")
+		os.Exit(1)
+	}
+
+	// TODO: determine how we can use mgr.GetClient. Currently the
+	//       manager's client doesn't recognize core objects.
+	k8sCoreClient, _ := client.New(mgr.GetConfig(), client.Options{})
+
+	kd, err := observer.GetDeployment(kdName, k8sCoreClient)
+	if err != nil {
+		log.Error(err, "failed to get kubedirector deployment object")
+		os.Exit(1)
+	}
+
+	// TODO: migrate the current webhook code to the controllers and using the
+	//       new framework; i.e. mgr.Add(hookServer) and hookServer.Register()
```

Comment:
I think we can fairly create an Issue for this, and delete the TODO comment.

Reply:
Done. Please let me know if the issue that I created is ok.